src/gateways/types.ts

```diff
@@ -1,67 +1,92 @@
-interface GatewayAddressRegistryItemData {
+export interface PaginatedResult<T> {
+  hasMore: boolean;
+  items: T[];
+  limit: number;
+  nextCursor: string;
+  sortBy: string;
+  sorOrder: string;
+  totalItems: number;
+}
+
+export interface GatewaySettings {
+  port: number;
+  protocol: string;
+  minDelegatedStake: number;
+  fqdn: string;
+  delegateRewardShareRatio: number;
+  autoStake: boolean;
+  note: string;
+  allowDelegatedStaking: boolean;
+  label: string;
+  properties:
+    | string
+    | {
+        GRAPHQL: boolean;
+        ARNS: boolean;
+        MAX_PAGE_SIZE: number;
+      };
+}
+
+export interface GatewayStats {
+  failedConsecutiveEpochs: number;
+  observedEpochCount: number;
+  passedConsecutiveEpochs: number;
+  totalEpochCount: number;
+  prescribedEpochCount: number;
+  passedEpochCount: number;
+  failedEpochCount: number;
+}
+
+export interface GatewayWeights {
+  compositeWeight: number;
+  observerRewardRatioWeight: number;
+  normalizedCompositeWeight: number;
+  tenureWeight: number;
+  gatewayRewardRatioWeight: number;
+  stakeWeight: number;
+}
+
+export interface GatewayAddressRegistryItemData {
+  gatewayAddress: string;
+  observerAddress: string;
   operatorStake: number;
-  vaults: GatewayVault[];
-  settings: {
-    label: string;
-    fqdn: string;
-    port: number;
-    protocol: string;
-    properties?: any;
-    note?: string;
-  };
-  status: "joined";
-  start: number;
-  end: number;
+  startTimestamp: number;
+  status: "joined" | "leaving";
+  totalDelegatedStake: number;
+  settings: GatewaySettings;
+  stats: GatewayStats;
+  weights: GatewayWeights;
 }
 
-export interface GatewayAddressRegistryItem
-  extends GatewayAddressRegistryItemData {
-  id: string;
-  linkFull: string;
-  linkDisplay: string;
-  ping: any;
-  health: any;
+export interface GatewayUnknownCheck {
+  status: "unknown";
+}
+
+export interface GatewayPendingCheck {
+  status: "pending";
 }
 
-export interface GatewayAddressRegistryCache {
-  // contractTxId: string;
-  gateways: Record<string, GatewayAddressRegistryItemData>;
```

Comment:
Types were mostly incorrect, incomplete or inaccurate. For instance, this structure never actually existed.